source/adapters/level_zero/image.cpp

```diff
@@ -114,32 +775,149 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesImageCopyExp(
     ur_rect_offset_t dstOffset, ur_rect_region_t copyExtent,
     ur_rect_region_t hostExtent, uint32_t numEventsInWaitList,
     const ur_event_handle_t *phEventWaitList, ur_event_handle_t *phEvent) {
-  std::ignore = hQueue;
-  std::ignore = pDst;
-  std::ignore = pSrc;
-  std::ignore = pImageFormat;
-  std::ignore = pImageDesc;
-  std::ignore = imageCopyFlags;
-  std::ignore = srcOffset;
-  std::ignore = dstOffset;
-  std::ignore = copyExtent;
-  std::ignore = hostExtent;
-  std::ignore = numEventsInWaitList;
-  std::ignore = phEventWaitList;
-  std::ignore = phEvent;
-  urPrint("[UR][L0] %s function not implemented!\n", __FUNCTION__);
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  std::scoped_lock<ur_shared_mutex> Lock(hQueue->Mutex);
```

Comment:
Should we try to share some code with `enqueueMemImageCommandHelper` in memory.cpp?

Reply:
why cant we use Image function   to copy from HOST  to Bindless Image ?

ze_image_handle_t  returned from zeImageCreate should work with standard image API

Reply:
if the underlying memory is non-usm, i.e. created by zeImageCreate, zeCommandListAppendImageCopyFromMemory is called as shown below, so it is image function.
I was intending to use zeCommandListAppendImageCopyFromMemoryExt/zeCommandListAppendImageCopyToMemoryExt, but then find they are not implemented in level-zero runtime. Is this a gap? @HoppeMateusz